### PR TITLE
SY Chou, CW Lin update Timer

### PR DESCRIPTION
**Pull request by FrankCCCCC**
Add 2 methods "setTimer", "recordTime" to incorporate with the modification in ElaSQL

### CODE REVIEW

File: src/main/java/org/vanilladb/core/util/Timer.java [hunk 1]
```diff
@@ -73,6 +73,12 @@ public void stopTimer() {
 				totalTime += (System.nanoTime() - start) / 1000;
 		}
 
+		// MODIFIED: Add method to set timer's time directly
+		public void setTimer(long timeInterval){
+			totalTime += timeInterval;
+			count++;
+		}
```
Comment (xuan19970825) [inline src/main/java/org/vanilladb/core/util/Timer.java:80]: It's better to use 'setTime' as function name.

File: src/main/java/org/vanilladb/core/util/Timer.java [hunk 2]
```diff
@@ -131,6 +137,17 @@ public long getComponentTime(Object component) {
...
+	// MODIFIED: Add method to record given time
+	public void recordTime(Object component, long time) {
+		SubTimer timer = subTimers.get(component);
+		if (timer == null) {
+			timer = new SubTimer();
+			timer.setTimer(time);
+			subTimers.put(component, timer);
+			componenents.add(component);
+		}
+	}
```
Comment (xuan19970825) [inline src/main/java/org/vanilladb/core/util/Timer.java:149]: Please use `timeInterval`  as the function input. 
Line 145 should be put at the end of this function.
```java
public void recordTime(Object component, long timeInterval) {
	SubTimer timer = subTimers.get(component);
	if (timer == null) {
		timer = new SubTimer();
		subTimers.put(component, timer);
		componenents.add(component);
	}
	timer.setTime(timeInterval);
	}
```
And please move this method to before `getComponentTime`.

**Review by xuan19970825 — COMMENTED** [review]
Remember to remove comments !

**Comment by SLMT** [comment]
Closed due to the update of ElaSQL.

**Pull request closed without merging**